### PR TITLE
Move the actions on the main page to a contextual menu

### PR DESCRIPTION
**Pull request by danidoni**
The actions present under the Welcome section take precious space on mobile
layouts. We move them under a contextual menu that pops up from the bottom of
the screen.

Those actions go back to its original place on bigger screens as there's
plenty of space.

Here is a screenshot of how it looks:

**Before**
_Small devices layout_
image

_Medium and bigger devices layout_
image

**After**
_Small devices layout_
image

_Medium and bigger devices layout_
image

Co-authored-by: David Kang <dkang@suse.com>

### CODE REVIEW

File: src/api/app/views/webui/main/_proceed-list.html.haml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-- if User.admin_session?
-  = render partial: 'admin_proceed-list'
-= render partial: 'user_proceed-list'
-- unless @spider_bot
-  .col
-    = link_to(monitor_path, class: 'btn btn-default', title: 'See the status of the builds') do
+.contextual-actions.d-none.d-md-flex
```
[the remark this answers is not in the document]
Comment (DavidKang) [reply src/api/app/views/webui/main/_proceed-list.html.haml]: Other side effect is that these changes also out of `beta`.

### NO BETA

In small viewports, the action are hidden.
Screenshot_2020-02-07 Open Build Service(2)

Also de disposition and style of the elements has changed

Screenshot_2020-02-07 Open Build Service(3)

File: src/api/app/views/layouts/webui/_proceed-list.html.haml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+HANDER
```
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/layouts/webui/_proceed-list.html.haml]: Crap! I have to remove this now! :sweat_smile:

[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/layouts/webui/_proceed-list.html.haml]: Deleted the unused partial

File: src/api/app/views/webui/main/_proceed-list.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/webui/main/_proceed-list.html.haml]: > In some breakpoint, the elements overflow the card
> 
> Screenshot_2020-02-07 Open Build Service(1)

Fixed:
image

Working to fix the other bugs

File: src/api/app/views/layouts/webui/responsive_ux/_proceed-list.html.haml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.navbar-collapse.contextual-collapse.navbar-dark.d-block.d-md-none
```
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/layouts/webui/responsive_ux/_proceed-list.html.haml]: Fixed:
image

File: src/api/app/views/webui/main/_proceed-list.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/webui/main/_proceed-list.html.haml]: > Other side effect is that these changes affect also out of `beta`.
> ### NO BETA
> 
> In small viewports, the actions are hidden.
> Screenshot_2020-02-07 Open Build Service(2)
> 
Fixed, see above

[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/webui/main/_proceed-list.html.haml]: > Also de disposition and style of the elements has changed
> 
> Screenshot_2020-02-07 Open Build Service(3)

Also fixed, see above

**Comment by danidoni** [comment]
Rebased

**Comment by hennevogel** [comment]
I must say, I don't like it too much. This is getting stuffy and confusing...

Screenshot from 2020-02-07 17-18-16

**Comment by danidoni** [comment]
Rebased

> I must say, I don't like it too much. This is getting stuffy and confusing...

Why do you find it confusing? The icons maybe?

> Why another menu instead of adding the actions to the already existing one?

The idea behind this new button is to move actions that depend on the page you are looking at. Some pages have actions, like the Welcome page or the Project page, but some others don't. If the page has actions, the button appears, so you know you can do stuff in that and only that page  (contextual actions), if the page doesn't have any action, the button does not show up (this is still not happening, I'm working on it).

Adding those actions to the already existing menu will hide them, so you won't know those are present unless you check it.

**Comment by dmarcoux** [comment]
The icons alone don't really convey their purpose. I believe adding a bit of text could help.

**Comment by danidoni** [comment]
I think I'll close this and work on another PR moving the actions into the user menu

**Comment by DavidKang** [comment]
@danidoni, then let's close this PR :smile_cat:

**Pull request closed without merging**